numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -51,6 +51,18 @@
 #endif
 /**********************************************/
 
+/**
+ * Thin wrapper around print that ignores exceptions
+ */
+static void
+_safe_print(PyObject *obj)
+{
+    if (PyObject_Print(obj, stdout, 0) < 0) {
+        PyErr_Clear();
+        printf("<error during print>");
+    }
+}
+
```

Comment:
Needs to only be conditionally compiled to avoid `warning: ???_safe_print??? defined but not used`